ompi/mca/hook/comm_method/hook_comm_method_component.c

```diff
@@ -111,63 +132,53 @@ static int ompi_hook_comm_method_component_register(void)
-                                           MCA_BASE_VAR_SCOPE_READONLY,
-                                           &mca_hook_comm_method_enable_mpi_init);
-
     mca_hook_comm_method_enable_mpi_finalize = false;
-    (void) mca_base_component_var_register(&mca_hook_comm_method_component.hookm_version, "enable_mpi_finalize",
-                                           "Enable comm_method behavior on mpi_finalize",
-                                           MCA_BASE_VAR_TYPE_BOOL, NULL,
-                                           0, 0,
-                                           OPAL_INFO_LVL_3,
-                                           MCA_BASE_VAR_SCOPE_READONLY,
-                                           &mca_hook_comm_method_enable_mpi_finalize);
-
-    // User can set the comm_method mca variable too
-    int hook_comm_method = -1;
-    (void) mca_base_var_register("ompi", NULL, NULL, "comm_method",
-                                 "Enable comm_method behavior (1) mpi_init or (2) mpi_finalize",
-                                 MCA_BASE_VAR_TYPE_INT, NULL,
-                                 0, 0,
-                                 OPAL_INFO_LVL_3,
-                                 MCA_BASE_VAR_SCOPE_READONLY,
-                                 &hook_comm_method);
-
-    if( 1 == hook_comm_method ) {
-        mca_hook_comm_method_enable_mpi_init = true;
+    mca_base_var_enum_create_flag("ompi_comm_method", mca_hook_comm_method_modes, &mca_hook_comm_method_flags);
+    
+    ret = mca_base_var_register("ompi", "ompi", NULL, "comm_method",
+                                "Enable the communication protocol report: when MPI_INIT is invoked (using the 'mpi_init' value) and/or when MPI_FINALIZE is invoked (using the 'mpi_finalize' value).",
+                                MCA_BASE_VAR_TYPE_UNSIGNED_INT,
+                                &mca_hook_comm_method_flags->super,
+                                0, 0,
+                                OPAL_INFO_LVL_3,
+                                MCA_BASE_VAR_SCOPE_READONLY,
+                                &mca_hook_comm_method_enabled_flags);
+    OBJ_RELEASE(mca_hook_comm_method_flags);
+    if(OPAL_ERR_VALUE_OUT_OF_BOUNDS == ret) {
+        opal_output(0, "hook:comm_method: Warning invalid comm_method specified.");
     }
-    else if( 2 == hook_comm_method ) {
-        mca_hook_comm_method_enable_mpi_finalize = true;
+    else {
+        if( mca_hook_comm_method_enabled_flags & OMPI_HOOK_COMM_METHOD_INIT ) {
+            mca_hook_comm_method_enable_mpi_init = true;
+        }
+        if( mca_hook_comm_method_enabled_flags & OMPI_HOOK_COMM_METHOD_FINALIZE ) {
+            mca_hook_comm_method_enable_mpi_finalize = true;
+        }
     }
 
-    // comm_method_max
-    (void) mca_base_var_register("ompi", NULL, NULL, "comm_method_max",
+    // hook_comm_method_max
+    (void) mca_base_component_var_register(&mca_hook_comm_method_component.hookm_version, "max",
```

Comment:
Did you intend these changes to be part of this commit / PR?  It changes the MCA param name from `ompi_comm_method_max` to `hook_comm_method_max`.  That isn't described in the commit message at all.

Also, there's now a disparity between `ompi_comm_method` and `hook_comm_method_*` -- why are some in the OMPI level and some in the hook framework level?

Reply:
Yeah, I meant them to be there. It seemed odd to have a bunch of `ompi_` specific controls for this. We wanted the `ompi_comm_method` to be general - not tied to a component - for ease of use/reference. The others seemed to me to be fine-tuning which seems like a component thing.

That said, since we intend to hide it behind the CLI option in https://github.com/openpmix/prrte/pull/711 we could make them all component level MCAs. Do you have a preference?

Reply:
Yeah, I'm sorry to be so nit-picky, but having fewer one-off situations as a good thing.

How about a compromise -- because the PRRTE PR CLI option is good for (effectively) `mpirun`, but not for direct launch:

1. Make the MCA param be `hook_...whatever` (i.e., properly prefixed)
2. Make an alias MCA param to it that is just `ompi_comm_method`.  That alias can have a useful help message and/or comment in the code explaining why it's not `hook_...`.

Indeed, I imagine that that should be fairly "normal" for hook components: they're OMPI-wide functionality.  So having an `ompi_FOO` MCA param seems appropriate.  But It is probably at least worth mentioning as a comment in the code, yadda yadda yadda.

Thanks!

Reply:
How about the ompi level MCA being `ompi_display_comm` to match the discussion [here](https://github.com/openpmix/prrte/pull/711#issuecomment-756268334)?

Reply:
Works for me.

Reply:
Done. Let me know what you think.